common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -509,9 +552,146 @@ bool base_wlan_hal_dwpal::attach_ctrl_interface(int vap_id)
 
 bool base_wlan_hal_dwpal::process_nl_events()
 {
-    LOG(DEBUG) << __func__ << "isn't implemented by derived and shouldn't be called";
-    return false;
+    // process all NL events
+    if (!m_dwpal_nl_ctx) {
+        LOG(ERROR) << "Invalid WPA Control socket used for nl events (m_dwpal_ctx == nullptr)";
```

Comment:
That's not a WPA Control socket... It's a Netlink socket.

Reply:
fixed